plugins/main/bridge/bridge.go

```diff
@@ -323,30 +355,40 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 
 	if n.IsGW {
-		var firstV4Addr net.IP
+		var firstV4GWAddr net.IP
+		var firstV6GWAddr net.IP
 		for _, ipc := range result.IPs {
 			gwn := &net.IPNet{
 				IP:   ipc.Gateway,
 				Mask: ipc.Address.Mask,
 			}
-			if ipc.Gateway.To4() != nil && firstV4Addr == nil {
-				firstV4Addr = ipc.Gateway
-			}
-
-			if err = ensureBridgeAddr(br, gwn, n.ForceAddress); err != nil {
-				return err
+			switch {
+			case ipc.Gateway.To4() != nil && firstV4GWAddr == nil:
+				firstV4GWAddr = ipc.Gateway
+				err = ensureBridgeAddr(br, netlink.FAMILY_V4, gwn, n.ForceAddress)
+				if err != nil {
+					return fmt.Errorf("failed to set bridge addr: %v", err)
+				}
+				if err := ip.EnableIP4Forward(); err != nil {
+					return fmt.Errorf("failed to enable forwarding: %v", err)
+				}
+			case len(ipc.Gateway) == net.IPv6len && firstV6GWAddr == nil:
+				firstV6GWAddr = ipc.Gateway
+				err = ensureBridgeAddr(br, netlink.FAMILY_V6, gwn, n.ForceAddress)
+				if err != nil {
+					return fmt.Errorf("failed to set bridge addr: %v", err)
+				}
+				if err := ip.EnableIP6Forward(); err != nil {
+					return fmt.Errorf("failed to enable forwarding: %v", err)
+				}
```

Comment:
Ugh, I feel like there's gotta be a way to genericize these two cases.  I didn't look, but perhaps we can make ensureBridgeAddr() detect the family based on length, as long as we ensure addresses are canonicalized to 4 or 16 byte sizes?